### ISSUE TITLE
High CPU Usage

### ISSUE THREAD

**Issue by zsqy** [owncloud/core#40532]
### Steps to reproduce
1. Install the latest version (mine: 10.11.0.6)
2. Upload a lot of files (mine: more than 400k files, total of more than 100GB)
3. In less than 12 hours, CPU will spike up to 100%, or
4. Login to MariaDB, run the following command and it will take forever to return which causing the high CPU usage
> select fileid, name, user_id from oc_filecache fc join oc_mounts on storage = storage_id where parent in (select fileid from oc_filecache where storage in (select numeric_id from oc_storages where id like 'home::%' or id like 'object::user:%') and path = 'thumbnails') and not exists(select fileid from oc_filecache where fc.name = CAST(oc_filecache.fileid as CHAR(24))) and fc.fileid > '0' order by user_id, fileid limit 10;

This query is added in https://github.com/owncloud/core/commit/5298002bd776156fd3e699ffa3f621d57dabb2dd

### Expected behaviour
CPU shouldn't be 100%

### Actual behaviour
CPU 100% for more than 2 days, didn't come down

### Server configuration
Not important, default installation will do, I use MariaDB.

### Client configuration
Not important, default configuration will do

### How did I solve?
Issue gone after adding the index to the path column in oc_filecache table
```ALTER TABLE oc_filecache ADD INDEX path_index (path)```

**Comment by ho4ho** [comment]
Could be same by #40383 with improving by #40514 ?

**Comment by paolocar88** [comment]
I confirm that the solution suggested (ALTER TABLE oc_filecache ADD INDEX path_index (path)) works.
Is it safe or does it work because it broke something else?

**Comment by stefanost67** [comment]
I get 

_mysql> ALTER TABLE oc_filecache ADD INDEX path_index (path);
ERROR 1071 (42000): Specified key was too long; max key length is 3072 bytes_

on percona xtradb-cluster 5.7

**Comment by ho4ho** [comment]
@zsqy @paolocar88 Me not knowing if changing ownCloud database scheme good idea. Could cause problems by upgrades? Better waiting by ownCloud team to check / response.

**Comment by djjudas21** [comment]
I was going to report the same issue. I am seeing this on ownCloud 10.11.0 with MariaDB 10.6.8. These long-running queries never complete, and consume all CPU on the server.

```
MariaDB [owncloud]> show processlist;
+--------+----------+-------------------+----------+---------+--------+--------------+------------------------------------------------------------------------------------------------------+----------+
| Id     | User     | Host              | db       | Command | Time   | State        | Info                                                                                                 | Progress |
+--------+----------+-------------------+----------+---------+--------+--------------+------------------------------------------------------------------------------------------------------+----------+
| 415000 | owncloud | 10.1.106.10:40744 | owncloud | Query   | 161763 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 419228 | owncloud | 10.1.106.10:60044 | owncloud | Query   | 148681 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 422172 | owncloud | 10.1.106.10:38278 | owncloud | Query   | 138842 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 424990 | owncloud | 10.1.106.10:56782 | owncloud | Query   | 129482 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 428254 | owncloud | 10.1.106.10:34602 | owncloud | Query   | 118560 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 441185 | owncloud | 10.1.106.10:57470 | owncloud | Query   |  75362 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 445583 | owncloud | 10.1.106.10:40044 | owncloud | Query   |  62225 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 448525 | owncloud | 10.1.106.10:47278 | owncloud | Query   |  52385 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 451315 | owncloud | 10.1.106.10:59924 | owncloud | Query   |  43024 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 454579 | owncloud | 10.1.106.10:47818 | owncloud | Query   |  32104 | Sending data | select `fileid`, `name`, `user_id` from `oc_filecache` `fc`
join `oc_mounts` on `storage` = `storage |    0.000 |
| 465521 | owncloud | localhost         | owncloud | Query   |      0 | starting     | show processlist                                                                                     |    0.000 |
+--------+----------+-------------------+----------+---------+--------+--------------+------------------------------------------------------------------------------------------------------+----------+
11 rows in set (0.000 sec)
```

```
MariaDB [owncloud]> select version();
+----------------+
| version()      |
+----------------+
| 10.6.8-MariaDB |
+----------------+
1 row in set (0.000 sec)
```

```
MariaDB [owncloud]> select count(*) from oc_filecache;
+----------+
| count(*) |
+----------+
|   298598 |
+----------+
1 row in set (0.319 sec)
```

I ran `OPTIMIZE TABLE oc_filecache;` and this did not help. Happy to supply any more details if they would help.

**Comment by zsqy** [comment]
> @zsqy @paolocar88 Me not knowing if changing ownCloud database scheme good idea. Could cause problems by upgrades? Better waiting by ownCloud team to check / response.

No, I don’t know what the impact is. But I do know the index can be dropped easily. We can just drop it before doing any upgrade later. So far I have been using it for 4 days after applying the index, around 30 active users daily, 800k number of files, 400GB storage, no other issue observed.

This is the temporary workaround:
``` ALTER TABLE oc_filecache ADD INDEX path_index (path) ``

**Comment by ho4ho** [comment]
Yes but others could not know applying index and have problem later. :-/

**Comment by ho4ho** [comment]
Fix by #40514
